### DISCUSSION TITLE
Pull request simulation

### DISCUSSION THREAD

**Discussion by IT-VBFK** [pcottle/learnGitBranching#1057] [Ideas]
Hello :)

I have an idea for a new feature (that might be "not feasible"), but thought I'd put this in here if it's not.

Just a few words about the background and motivation of this request:
In my daily work we switched from SVN (yes, sorry to drop that bad word here ;) ) to git. Well, the real benefit of git is that it's really easy to collaborate on. But once we started using git, we're a bit overwhelmed by PRs and the "magic" that goes on behind the scenes.

I think if there had been a nice visualization of the process etc we would have taken that with more ease.

Simulating the process of a PR should contain IMHO
- forking an upstream repo
- clone the fork
- creating a feature in a separate branch (which is possible right now)
- push this branch (which is possible right now)
- create a pull/merge request (which is not possible)
- do some changes and (force) push inside this feature branch (which is possible right now)
- accept the PR and merge it in main/develop/whatever (which is not possible)

**Comment by pcottle** [comment]
ahhh interesting, so like a full-on github simulator? That could be cool to have one day for sure.

I actually think github itself used to have a very hands-on simulator experience that taught you git and pull requests, it lived at `try.github.io`, but sadly its gone now. an old archive of this:
https://web.archive.org/web/20150207185123/https://try.github.io/levels/1/challenges/1

I think it would be ideal for github to revisit building this, but I understand with their layoffs that its probably not a priority right now :/

**Comment by ITaluone** [comment]
Thinking out loud.. This could then include also working with multiple remotes :)

Just as a side note :)

**Comment by Magador** [comment]
During a Git training at work, my colleague showed us your tool and it was amaazing watching him explain the concepts visually. I was assisting him and during a break, we talked about the fact that it was not possible to simulate a GitLab MR or a GitHub PR with the tool.

I created the PR #1120 to emulate the behavior of merging the requests through their respective tools (even deleting the remote branch) and matching the Git behavior on the local side when fetching, or re-pushing a deleted branch with new work (or not).

**Reply by pcottle** [reply]
Thanks for the contribution @Magador ! I think this addresses a big chunk of the issue -- getting more familiarity with the workflow of merge requests. Hope this helps address your feedback @IT-VBFK

**Reply by IT-VBFK** [reply]
Yes I does :)
